### PR TITLE
update README.md

### PR DESCRIPTION
**Pull request by KamikX**
add Knowledge base - usage with third party libraries section with Select2 example

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -273,3 +273,96 @@ the HTML on the page.
 
 This is a non-trivial task and there may be room for improvement in this
 library to make this easier. If you have ideas, please open an issue!
+
+## Knowledge base - usage with third party libraries
+### Select2
+If you're using [Select2](https://select2.org/) that is instantiated on the page load, you may need to wrap it in a Stimulus controller to make it work when new fields dynamically pop up into your HTML DOM.  
+Example: [Symfony UX Turbo](https://symfony.com/bundles/ux-turbo/current/index.html#decomposing-complex-pages-with-turbo-frames) + [Custom stimulus controller](https://stimulus.hotwired.dev/reference/controllers)
+
+**_select2_controller.js_**
```
Comment (bocharsky-bw) [inline README.md:282]: No need in this as the file name along with the path is specified in the code block itself

File: README.md [hunk 2]
```diff
@@ -273,3 +273,96 @@ the HTML on the page.
...
+
+    connect() {
+        this.initBaseSelector(this.select)
+    }
+
+    disconnect() {
+        this.select.select2('destroy');
+    }
+
+    bindOnSelect2Event(select2) {
```
Comment (bocharsky-bw) [inline README.md:310]: I would move this method after `initBaseSelector()` for better readability

File: README.md [hunk 3]
```diff
@@ -273,3 +273,96 @@ the HTML on the page.
...
+
+/* stimulusFetch: 'lazy' */
+import {Controller} from "@hotwired/stimulus";
+
+export default class extends Controller {
+
+    static values = {
+        allowClear: {type: Boolean, default: false},
+        placeholder: {type: String, default: null},
+        closeOnSelect: {type: Boolean, default: true}
```
Comment (bocharsky-bw) [inline README.md:295]: Maybe group bool types together? Like:
```suggestion
        placeholder: {type: String, default: null},
        allowClear: {type: Boolean, default: false},
        closeOnSelect: {type: Boolean, default: true},
```

**Review by bocharsky-bw — COMMENTED** [review]
I think we should explain in short what that code does. I mean, *why* we need that code (and seems you explained it before the code block) and *what* this code should do? Should it trigger that "change event" on some specific select2 events?

**Comment by KamikX** [comment]
> Do you refer to any known issue or PR with these changes?

PR refer to this [issue](https://github.com/SymfonyCasts/dynamic-forms/issues/30#issuecomment-2357869879) and your suggestio.

**Comment by bocharsky-bw** [comment]
Thanks for sharing the cross-link! Now it will be easier to find your code snippet for others on GitHub 👍

**Pull request closed without merging**